### PR TITLE
[T3A3][T10-A4]ZavierOng

### PR DESCRIPTION
**Pull request by E0002878**

### CODE REVIEW

**Comment by E0002878** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -21,35 +25,62 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+    	
         this.isPrivate = isPrivate;
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        createAddress(address);
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: good job applying SLAP here.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -21,35 +25,62 @@
...
+    }
+    
+    private void createAddress(String address) throws IllegalValueException {
+    	final int BLOCK_INDEX = 0;
+    	final int STREET_INDEX = 1;
+    	final int UNIT_INDEX = 2;
+    	final int POSTAL_CODE_INDEX = 3;
+
+        String[] addressSegment = address.split(",");
+        if (addressSegment.length != 3) {
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: addressSegment should have 4 parts right? 3 commas.

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a person's block in the addressbook
+ * @author ZavierOng
+ */
+public class Block {
+	
+	public static final String EXAMPLE = "123";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS =
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Block.java]: do take note of indentation

**Comment by mauris** [comment]
Some comments added. Feel free to comment if you have any questions. Once done, you may the PR after reading comments.

**Pull request closed without merging**